R/send_sparql.R

```diff
@@ -12,38 +12,57 @@
 #'} ORDER BY ?itemLabel
 #''
 #'send_sparql(query=metro_query)
-#' @import SPARQL
 send_sparql=function(query,endpoint="Wikidata"){
   endpoint=tolower(endpoint)
 
   # if endpoint wikidata, use WikidataQueryServiceR::query_wikidata()
   if(endpoint=="wikidata"){
-    tib <- purrr::quietly(WikidataQueryServiceR::query_wikidata)(query)$result
+    return(purrr::quietly(WikidataQueryServiceR::query_wikidata)(query)$result)
   }
-  # else, use SPARQL::SPARQL()
+  # else, use httr2
+
   # if endpoint passed as name, get url
-  if(endpoint %in% (usual_endpoints %>%
-                    dplyr::filter(.data$name!="wikidata") %>%
-                    dplyr::pull(.data$name))){
-    url=usual_endpoints %>%
-      dplyr::filter(.data$name == {{ endpoint }}) %>%
-      dplyr::select(.data$url) %>%
-      dplyr::pull()
-    tib <- SPARQL::SPARQL(url=url,
```

Comment:
In practice this uses the default `format` value for `SPARQL::SPARQL()` which is "xml", but this gets a data.frame. :raised_eyebrow: https://github.com/cran/SPARQL/blob/51b993f16cb924f2ccc325f08f42da3420ea08bc/R/SPARQL.R#L63

Reply:
That argument was more for the HTTP header than for the output I suppose.

Reply:
Found my favorite part of parsing in {SPARQL}: https://github.com/cran/SPARQL/blob/51b993f16cb924f2ccc325f08f42da3420ea08bc/R/SPARQL.R#L141 :grin: